src/views/List.jsx

```diff
@@ -1,15 +1,44 @@
 import { ListItem } from '../components';
+import { useState } from 'react';
 
 export function List({ data }) {
+	const [searchInput, setSearchInput] = useState('');
+
 	const listItemsToDisplay = data.map((item) => {
-		return <ListItem key={item.id} name={item.name} />;
+		return item.name?.toLowerCase().includes(searchInput.toLowerCase()) ? (
+			<ListItem key={item.id} name={item.name} />
+		) : null;
```

Comment:
Even though this works fine, I have to spend extra brain cycles thinking about "what is this piece of code doing here? 🤔", instead, I'd suggest you extracting

```js
item.name?.toLowerCase().includes(searchInput.toLowerCase())

// into a constant/function like this
const isItemInSearchCritera = item.name?.toLowerCase().includes(searchInput.toLowerCase());

return isItemInSearchCriteria ? <ListItem .../> : null
```

I understand that `isItemInSearchCriteria` is probably not the best name for a constant, but at least it makes the intent of the code clearer!

Having readable code >>> Having smart code

Reply:
For what is worth, this is not a blocker, feel free to leave it like this if you want.

Reply:
And the string normalization (make everything `lowerCase`) was on point, great work! 💯